helpful_tools/Install-DockerCE/uninstall-docker-ce.ps1

```
@@ -0,0 +1,645 @@
+                & icacls.exe $global:DockerDataPath /grant "$env:USERNAME`:F" /t /c 2>$null | Out-Null
+                
+                # Now attempt to remove the directory
+                Remove-Item $global:DockerDataPath -Recurse -Force
+                Write-Output "Docker data directory removed."
+            }
+            catch
+            {
+                Write-Warning "Failed to remove Docker data directory: $_"
+                Write-Warning "You may need to manually remove $global:DockerDataPath"
+            }
+        }
+        else
+        {
+            Write-Output "Docker data directory not found."
+        }
+        
+        # Also remove downloaded Docker files from user profile
+        $dockerDownloads = "$env:UserProfile\DockerDownloads"
+        if (Test-Path $dockerDownloads)
+        {
+            Write-Output "Removing Docker download files..."
+            try
+            {
+                Remove-Item $dockerDownloads -Recurse -Force
+                Write-Output "Docker download files removed."
+            }
+            catch
+            {
+                Write-Warning "Failed to remove Docker download files: $_"
+            }
+        }
+    }
+    else
+    {
+        Write-Output "Preserving Docker data directory as requested."
+    }
+}
+
+function
+Remove-DockerRegistryKeys()
+{
+    Write-Output "Removing Docker registry keys..."
+    
+    $registryPaths = @(
+        "HKLM:\SYSTEM\CurrentControlSet\Services\docker",
+        "HKLM:\SYSTEM\ControlSet002\Services\docker"
+    )
+    
+    foreach ($regPath in $registryPaths)
+    {
+        try
+        {
+            if (Test-Path $regPath)
+            {
+                Write-Output "Removing registry key: $regPath"
+                Remove-Item $regPath -Recurse -Force
+                Write-Output "Registry key removed: $regPath"
+            }
+            else
+            {
+                Write-Output "Registry key not found: $regPath"
```

Don't need to output if not found. Silently continue. We don't need user to observe our checks for this.